### PR TITLE
Merge pull request #1193 from guardian/gtrufitt/formatAttrName

### PR DESCRIPTION
**Pull request by gtrufitt**
## What does this change?

Adds a util to take a string and sanitise it a bit. May have over-thought/duplicated existing functionality, so shout if I have...

## Why?

We have a requirement to take `string => string formatted and hyphenated`.

### CODE REVIEW

File: src/web/lib/formatAttrString.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    const removeAlphaNumeric: strInOutFunc = str =>
+        str.replace(new RegExp('[^0-9a-z ]', 'gi'), ';'); // We set it to semicolon so we can remove it at the end
+    const lowercase: strInOutFunc = str => str.toLowerCase();
+    const replaceSpacesWithHyphens: strInOutFunc = str =>
+        str.replace(new RegExp(' ', 'g'), '-'); // Replace the space
+    const removeTempChars: strInOutFunc = str =>
+        str.replace(new RegExp('[;]', 'g'), ''); // Strip out those semicolons
+
+    // https://lodash.com/docs/4.17.15#flow
```
Comment (oliverlloyd) [inline src/web/lib/formatAttrString.ts]: Lodash flow is 1kb: https://bundlephobia.com/result?p=lodash.flow@3.5.0 so I'm aware I'm in the realm of nitpicking but maybe more ugly code is worth slightly less bytes? I don't have strong feelings on this one though!

Comment (gtrufitt) [reply src/web/lib/formatAttrString.ts]: Fair point.

There's a certain beauty in 

image

Comment (oliverlloyd) [reply src/web/lib/formatAttrString.ts]: Oh no... what have I done 🤦‍♂

Comment (gtrufitt) [reply src/web/lib/formatAttrString.ts]: too late! merged!

tenor-217053173

**Pull request merged**